diff -rupN dmidecode-2.12_orig/dmidecode.c dmidecode-2.12/dmidecode.c
--- dmidecode-2.12_orig/dmidecode.c	2013-04-17 08:25:34.000000000 -0400
+++ dmidecode-2.12/dmidecode.c	2015-05-04 07:01:00.853649954 -0400
@@ -69,7 +69,7 @@
 #define out_of_spec "<OUT OF SPEC>"
 static const char *bad_index = "<BAD INDEX>";
 
-#define SUPPORTED_SMBIOS_VER 0x0207
+#define SUPPORTED_SMBIOS_VER 0x0909
 
 /*
  * Type-independant Stuff
